testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/AbstractSSLMasterSlaveTestCase.java

```diff
@@ -0,0 +1,201 @@
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.READ_ATTRIBUTE_OPERATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+import static org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil.SLAVE_HOST_PASSWORD;
+import static org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil.SLAVE_HOST_USERNAME;
+import org.jboss.as.test.integration.domain.management.util.DomainTestSupport;
+import static org.jboss.as.test.integration.management.util.CustomCLIExecutor.MANAGEMENT_HTTP_PORT;
+import static org.jboss.as.test.integration.management.util.ModelUtil.createOpNode;
+import org.jboss.as.test.integration.security.common.CoreUtils;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+
+/**
+ * Common base for tests of SSL secured communication between master and slave.
+ *
+ * @author Ondrej Kotek <okotek@redhat.com>
+ */
+public abstract class AbstractSSLMasterSlaveTestCase {
+
+    protected static final String MASTER_MANAGEMENT_REALM = "MasterManagementRealm";
+    private static final String MGMT_CTX = "/management";
+    private static final int TIMEOUT = 60000;
+
+
+    protected static void keyMaterialSetup(File workDir) throws Exception {
+        // create key and trust stores with imported certificates from opposing sides
+        FileUtils.deleteDirectory(workDir);
+        workDir.mkdirs();
+        Assert.assertTrue(workDir.exists());
+        Assert.assertTrue(workDir.isDirectory());
+        CoreUtils.createKeyMaterial(workDir);
+    }
+
+    protected static void setMasterManagementNativeInterfaceAndCheck(ModelControllerClient client) throws Exception {
+        checkHostStatusOnMaster("slave");
+        setMasterManagementNativeInterface(client);
+        reloadMaster();
+        checkHostStatusOnMaster("master");
+    }
+
+    protected static void setOriginMasterManagementNativeInterfaceAndCheck() throws Exception {
+        setOriginMasterManagementNativeInterface();
+        reloadMaster();
+        checkHostStatusOnMaster("master");
+    }
+
+    protected static void reloadMaster() throws Exception {
+        ModelNode op = new ModelNode();
+        op.get(OP).set("reload");
+        op.get(OP_ADDR).add(HOST, "master");
+        op.get(ADMIN_ONLY).set(false);
+        executeOverHttp(getMasterMgmtUri(), op.toJSONString(true));
+    }
+
+    protected static void checkHostStatusOnMaster(String host) throws Exception {
+        final URI mgmtURI = getMasterMgmtUri();
+        final String operation = getHostStateRunningOperationJson(host);
+        final long time = System.currentTimeMillis() + TIMEOUT;
+        do {
+            Thread.sleep(1000);
```

Comment:
Please move the sleep below the "if (isHostStateRunning(mgmtURI, operation)) {" block and decrease the sleep time to 100ms at most. The testsuite should run fast.